src/landpatterns/courtyard.stanza

```diff
@@ -62,8 +62,7 @@ public defn compute-courtyard-boundary (
   val pkg-outline = bounds(envelope(pkg-body, density-level = density-level))
   val outline = union(pad-outline, pkg-outline)
   val outline* = fatten(excess, outline)
-  pos * LineRectangle(outline*, line-width = DEF_COURTYARD_WIDTH)
```

Comment:
perhaps getting rid of the DEF_COURTYARD_WIDTH constant would be a good idea at some point? Or perhaps it's needed for things that aren't the outline? The width of the outline can technically be controlled through UI styles.

Reply:
I can at least make it more specific to the origin marker.

Reply:
is it common practice to have the origin marker on the courtyard layer? perhaps we need to be smarter in the UI for stuff like that (e.g. for alignment)?

Reply:
It is pretty common in my experience. The default IPC footprint generator in Altium will create an origin marker like this. I tend to like having them for components where the land pattern center is not the same as the "center of mass" of the component.

Reply:
Ideally we'd create our landpatterns so their origin is the actual "origin" that we want, same for modules, but I don't know if that's always feasible especially for imports.

Reply:
Unfortunately it isn't always that easy. For example - see this part:
https://productfinder.pulseelectronics.com/api/open/part-attachments/datasheet/JD0-0001NL

![Screenshot 2024-07-26 at 12 10 05 PM](https://github.com/user-attachments/assets/3faaaa65-88bb-44fa-a803-0eda815f8193)

Notice how the mechanical drawing references all of the dimensions from the mechanical mounting hole as a datum ? 

In legacy CAD - what you will often do is either create the landpattern in that shift frame of reference and then re-set the origin once you are done or  you will just leave it in the shifted frame of reference. 

In the landpattern framework in JSL - I have an implementation that allows me to create a child node in that shift frame of reference. The JSL approach is far superior in my opinion because I can work in the shifted frame of reference but keep the origin of the landpattern at the components center of mass.

Reply:
Yeah, makes sense. We don't really care for the details of the land pattern construction at all, it's the components frame of reference that's relevant for things like placement and what not. I think what I'm saying is what you're also saying, that it would be cool if the "origin marker" isn't just something in a courtyard layer, but the actual origin of either module or component, I feel like that'll make things easier in the long run.